pkg/satellite/stream/stream.go

```diff
@@ -296,6 +326,9 @@ func (ss *satelliteStream) recvLoop() {
 				}
 				payload := telemetry.Data
 				log.Debug("received data: streamId: %v, planId: %s, framing type: %s, size: %d bytes\n", ss.streamId, planId, telemetry.Framing, len(payload))
+				if ss.enableAutoClose && len(payload) == 0 {
+					streamEndDetected = true
```

Comment:
I wonder if there is a specific reason why a boolean value is updated and checked every second (L203); could `ss.performAutoClose()` be called directly?